make `itk.ants_registration` have the same API and behavior as `ants.registration`

Ref: https://antspyx.readthedocs.io/en/latest/registration.html

The wrapping of itk.ANTSRegistration, `itk.ants_registration`, should have a similar API and behavior as antspy `ants.registration`.

First add tests following the ANTs tests for just all the C++ interfaces, options and behavior.

Then add the corresponding Python tests.

My WIP is coming up with:
```log
CMake Error at CMakeLists.txt:39 (find_package):
  By not providing "FindANTS.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "ANTS", but
  CMake did not find one.

  Could not find a package configuration file provided by "ANTS" with any of
  the following names:

    ANTSConfig.cmake
    ants-config.cmake
```
The easiest path forward seems to be to resolve https://github.com/ANTsX/ANTs/issues/1369.

--- a/include/itkANTSRegistration.hxx
+++ b/include/itkANTSRegistration.hxx
@@ -488,6 +488,13 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::GenerateData(
   {
     SingleStageRegistration(xfrmMethod, initialTransform, fixedImage, movingImage, true);
   }
+  else if (whichTransform == "quickrigid")
+  {
+    auto originalIterations = m_AffineIterations;
+    m_AffineIterations = { 20, 20, 0, 0 };
+    SingleStageRegistration(RegistrationHelperType::XfrmMethod::Rigid, initialTransform, fixedImage, movingImage, true);
+    m_AffineIterations = originalIterations;
+  }
   else if (whichTransform == "trsaa")
   {
     auto originalGradientStep = m_GradientStep;
@@ -536,6 +543,21 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::GenerateData(
     intermediateTransform = m_Helper->GetModifiableCompositeTransform();
     SingleStageRegistration(
       RegistrationHelperType::XfrmMethod::SyN, intermediateTransform, fixedImage, movingImage, true);
+  }
+  else if (whichTransform == "syncc")
+  {
+    std::string originalMetric = m_AffineMetric;
+    m_AffineMetric = "CC";
+    SingleStageRegistration(
+      RegistrationHelperType::XfrmMethod::Affine, initialTransform, fixedImage, movingImage, m_MaskAllStages);
+    m_AffineMetric = originalMetric;
+    this->UpdateProgress(0.15);
+    originalMetric = m_SynMetric;
+    m_SynMetric = "CC";
+    typename OutputTransformType::Pointer intermediateTransform = m_Helper->GetModifiableCompositeTransform();
+    SingleStageRegistration(
+      RegistrationHelperType::XfrmMethod::SyN, intermediateTransform, fixedImage, movingImage, true);
+    m_SynMetric = originalMetric;
   }
   else if (whichTransform.substr(0, 3) == "tv[") // TV[n]
   {

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -498,6 +498,38 @@ itk_add_test(NAME antsRegistrationTest_SyNScaleNoMasks_2stage
     1  # maskAllStages
   )
 
+itk_add_test(NAME antsRegistration_SyNCC
+  COMMAND ANTsWasmTestDriver
+    --compare
+    DATA{Baseline/antsRegistrationTest_SyNScaleNoMasks_Float.result.nii.gz}
+    ${ITK_TEST_OUTPUT_DIR}/antsRegistration_SyNCC.result.nii.gz
+    --compareIntensityTolerance 9
+    --compareRadiusTolerance 1
+    --compareNumberOfPixelsTolerance 1000
+  itkANTSRegistrationTest
+    DATA{Input/test.nii.gz}  # fixed image
+    DATA{Input/scale.test.nii.gz}  # moving image
+    ${ITK_TEST_OUTPUT_DIR}/antsRegistration_SyNCC.tfm  # output transform
+    ${ITK_TEST_OUTPUT_DIR}/antsRegistration_SyNCC.result.nii.gz  # moving image warped to fixed space
+    DATA{Input/Initializer_0.05_antsRegistrationTest_AffineScaleMasks_Float.mat}  # initial transform
+    none  # fixedMask
+    none  # movingMask
+    0.25  # GradientStep
+    SyNCC
+    Irrelevant  # affineMetric
+    0.20  # samplingRate
+    4  # ccRadius
+    25x20x5  # affineIterations
+    3x2x1  # shrinkFactors
+    2x1x0 # smoothingSigmas
+    0  # randomSeed (0 means do not set)
+    Irrelevant  # synMetric
+    100x70x20  # synIterations
+    --float
+    0  # collapseTransforms
+    1  # maskAllStages
+  )
+
 itk_add_test(NAME antsRegistration_SyNRA
   COMMAND ANTsWasmTestDriver
     --compare

